IntegrationTests/Services/AWSBedrockRuntimeIntegrationTests/BedrockAPIKeyGenerator.swift

```diff
@@ -0,0 +1,106 @@
+    ///   - awsCredentialIdentityResolver: The AWS credential identity resolver to use when resolving the AWS credentials
+    ///   that are used to sign the key.  Defaults to the AWS SDK for Swift's default identity resolver chain.
+    init(
+        region: String,
+        duration: TimeInterval = 43200.0,  // 12 hrs
+        awsCredentialIdentityResolver: any AWSCredentialIdentityResolver =
+            DefaultAWSCredentialIdentityResolverChain()
+    ) {
+        self.region = region
+        self.duration = duration
+        self.awsCredentialIdentityResolver = awsCredentialIdentityResolver
+    }
+    
+    /// Resolves AWS credentials, then uses those to create a new Bedrock API Key.
+    ///
+    /// The Bedrock API Key will be configured with the signing region & duration that were configured at
+    /// token generator creation.
+    /// - Returns: A `String` bearer token suitable for use with AWS Bedrock.
+    func generate() async throws -> String {
+
+        // Create a request builder to be signed as a presigned URL in the form:
+        // POST https://bedrock.amazonaws.com/?Action=CallWithBearerToken
+        // host header is set because it is required to be included in signature when presigning a URL
+        let inputBuilder = HTTPRequestBuilder()
+            .withMethod(.post)
+            .withHost(host)
+            .withHeaders(Headers(["host": host]))
+            .withQueryItem(URIQueryItem(name: "Action", value: "CallWithBearerToken"))
+
+        // Create signing properties for presigning a URL
+        var signingProperties = Attributes()
+        signingProperties.set(key: SigningPropertyKeys.bidirectionalStreaming, value: false)
+        signingProperties.set(key: SigningPropertyKeys.unsignedBody, value: false)
+        signingProperties.set(key: SigningPropertyKeys.signingName, value: signingName)
+        signingProperties.set(key: SigningPropertyKeys.signingRegion, value: region)
+        signingProperties.set(key: SigningPropertyKeys.signingAlgorithm, value: .sigv4)
+        signingProperties.set(key: SigningPropertyKeys.signatureType, value: .requestQueryParams)
+        signingProperties.set(key: SigningPropertyKeys.expiration, value: duration)
+
+        // Resolve the AWS credentials to be used for presigning
+        let awsCredentialIdentity = try await awsCredentialIdentityResolver
+            .getIdentity(identityProperties: Attributes())
+
+        // Perform the URL presigning, a new request builder is returned with signature applied
+        CommonRuntimeKit.initialize()
+        let signedRequestBuilder = try await SigV4Signer().signRequest(
+            requestBuilder: inputBuilder,
+            identity: awsCredentialIdentity,
+            signingProperties: signingProperties
+        )
+
+        // Create a presigned URL
+        let signedRequest = signedRequestBuilder.build()
+        let url = signedRequest.endpoint.uri.url!
```

Comment:
note: `!` won't ever fail here because the essential URL components were set on the `inputBuilder` above